### PR TITLE
Lavaland / Lavaplanet is now visible from space

### PR DESCRIPTION
**Pull request by ghost**
:cl: Keekenox & Floyd / Qustinnus
imageadd: Adds a new Parallax layer that resembles Lavaland (Lava Planet), it spawns on a random location near the station. You need your parallax on high to see it.
/:cl:

credits to the talented Keekenox for the sprites.

This PR adds a Lava planet to the parallax, its position is a bit more "absolute" than the other parallax, its located at around the center of the map, and will move more out of view the more you go to the edge of the zlevel

image
full planet

image
image
image
pics in station

image

near edge of Z-level

https://i.gyazo.com/0042993502311c22b2003f73d5cafeb9.mp4
vid ingame

### CODE REVIEW

**Comment by Errorin** [comment]
That's pretty sick actually. Nice work to keek.

**Review by Tlaltecuhtli — CHANGES_REQUESTED** [review]
not 3/4

**Comment by Tacolizard** [comment]
really cool, only complaint is that it looks just a little flat. The moon needs more shading or something I think

**Comment by fludd12** [comment]
Oh my god this is amazing.

**Comment by PKPenguin321** [comment]
so it's always visible no matter what provided you can see space?

bit iffy on that

**Comment by Qustinnus** [comment]
@PKPenguin321 no because it's centered so it'll slowly move more and more out of view, look at the last pic

**Comment by Slignerd** [comment]
What happens once you end up on another Z-level?

**Comment by PKPenguin321** [comment]
but... it's still in view in that last pic

**Comment by Qustinnus** [comment]
Currently it stays on screen, I can optionally remove it from screen when you move off of main Zlevel, but I might need some help implementing that properly.

**Comment by Wilchenx** [comment]
Nya?

**Comment by Qustinnus** [comment]
@PKPenguin321 
image


you can keep going and going further away. this isn't even at the end of the Zlevel.

**Comment by Qustinnus** [comment]
https://i.gyazo.com/82890d630987f0d6281dc56a6f2da768.mp4

for a video

**Comment by PKPenguin321** [comment]
maybe it should not be so centered. imo the pics of the windows near the bridge are much uglier than just space. the sprite itself is nice but when youre only seeing a fifth of it through a window it just looks like a red blob

**Comment by Tacolizard** [comment]
yeah maybe move it a bit more off-center

**Comment by Iamgoofball** [comment]
it's fine as-is tbh

**Comment by Robustin** [comment]
Came to here to say it shouldn't be sitting near the center, looks like im not alone.

**Comment by GunHog** [comment]
Could the position be randomized?

**Comment by Qustinnus** [comment]
yes it could

**Comment by Qustinnus** [comment]
i think

**Comment by epochayur** [comment]
this looks awesome but please make it randomized

**Comment by Qustinnus** [comment]
It's now randomized.

**Comment by nicbn** [comment]
This is too vibrant and too shiny, make it darker and lower the saturation

**Comment by nicbn** [comment]
Also if I'm not wrong the station canonically orbits a gas giant and lavaland is just a nearby planet or moon

**Comment by praisenarsie** [comment]
hi i'm a coder and what is lore?

**Comment by Qustinnus** [comment]
@nicbn read my response to slig

for now, lets just say our station is under the gas giant and we cant see it :^))

I'll look if someone is interested in making a gas giant sprite, but that'll probably be a seperate PR because the sprite would be a little smaller than 480x480 which is quite a lot of work.

**Comment by Qustinnus** [comment]
image
this is 75% saturation, I don't really have an eye for art so I'm not sure if this looks better. honestly I'm just waiting for @WJohn to say what he thinks

**Comment by ChuckTheSheep** [comment]
It looks scaled to 4x, what would it look like at 2x or 1?

**Comment by Iamgoofball** [comment]
It's not upscaled iirc.

**Comment by ChuckTheSheep** [comment]
Maybe not via programming, but per pixel it looks at least 2x

**Comment by Qustinnus** [comment]
The pixels are 2x, but I didn't scale it. Also, making it smaller ingame would look bad imo

**Comment by praisenarsie** [comment]
fractured moon zlevel when?

**Comment by praisenarsie** [comment]
BRING BACK MOONOUTPOST19

**Comment by nicbn** [comment]
75% looks better

**Comment by Qustinnus** [comment]
@nicbn that would work untill you realize that parallax is a HUD icon

**Comment by optimumtact** [comment]
here at /tg/station we don't let lore get in the way of the game

**Comment by Qustinnus** [comment]
thanks gamer

**Comment by nicbn** [comment]
But dude it's a fucking gas giant
We could use it for other features like gas mining or something

**Comment by epochayur** [comment]
yeah, we could, but is anyone gonna? this is cool in the meantime

**Review by WJohn — DISMISSED** [review]
Yup. Though the quality and style of the planet versus the background are very clearly different, this is such a cool idea and the sprite looks nice enough on its own that I absolutely approve of this. Perhaps someone in the future can give this thing a polishing to make it more in line with the background's style.

**Comment by Qustinnus** [comment]
YASSSS GUUURLL SLAAAAAYY

File: code/_onclick/hud/parallax.dm [hunk 1]
```diff
@@ -194,18 +196,27 @@
...
-			L.offset_y -= 480
-		if(L.offset_y < -240)
-			L.offset_y += 480
+
+		var/change_x
+		var/change_y
+
+		if(L.absolute)
+			L.offset_x = -(posobj.x - SSparallax.planet_x_offset) * L.speed
+			L.offset_y = -(posobj.y - SSparallax.planet_y_offset) * L.speed
```
Comment (ninjanomnom) [inline code/_onclick/hud/parallax.dm]: Make a new /obj/screen/parallax_layer proc that does the normal behavior then override it in the planet version.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/_onclick/hud/parallax.dm (hunk 1 above)
Comment (optimumtact) [reply code/_onclick/hud/parallax.dm]: proc overhead

**Comment by WJohn** [comment]
Main thing that's bothering me about this is the 2x scaling. I know that shrinking it down would make it look very small, but I think it'd be better as it would at least not clash with the pixel resolution of the station and the space background behind it.

Either that or double down on its resolution, which would be a lot of work.

File: code/_onclick/hud/parallax.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/_onclick/hud/parallax.dm]: If we're going to be having parallax layers act differently, putting the different behaviors in the generic loop seems a bad move

Comment (optimumtact) [reply code/_onclick/hud/parallax.dm]: This is something that can be applied to other parallax layers, and the behaviour would be useful. The argument for including it as a var instead makes more sense IMO and parallax loops need to be fairly performing

**Comment by KorPhaeron** [comment]
Why are people so obsessed with a gas giant that has never existed outside of a line of wiki text a random player added

**Comment by Qustinnus** [comment]
muh plasmeme and muh reason to be here

**Comment by Qustinnus** [comment]
image
image
image
image
@WJohn

**Comment by epochayur** [comment]
oh baby

**Review by WJohn — DISMISSED** [review]
Thanks

**Comment by Bawhoppen** [comment]
:+1: x :100:

File: code/_onclick/hud/parallax.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/_onclick/hud/parallax.dm]: That's fair enough, we can switch it over easily enough I guess if we end up needing to do this.

**Review by ninjanomnom — DISMISSED** [review]

File: code/controllers/subsystem/parallax.dm [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+	flags = SS_POST_FIRE_TIMING | SS_BACKGROUND
 	priority = FIRE_PRIORITY_PARALLAX
 	runlevels = RUNLEVEL_LOBBY | RUNLEVELS_DEFAULT
 	var/list/currentrun
+	var/planet_x_offset = 128
+	var/planet_y_offset = 128
+
+/datum/controller/subsystem/parallax/Initialize(timeofday)
+	planet_y_offset = rand(100, 160)
+	planet_x_offset = rand(100, 160)
```
Comment (ninjanomnom) [inline code/controllers/subsystem/parallax.dm]: You need to call parent

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Comment by banthebantz** [comment]
gas giant sprite and away missions next

**Comment by DaydreamPuppeh** [comment]
Will it appear on insane?

**Comment by Qustinnus** [comment]
Ya, hard and insane

**Pull request merged**